server-common/src/main/java/org/apache/kafka/server/group/share/DefaultStatePersister.java

```diff
@@ -259,7 +261,62 @@ public CompletableFuture<DeleteShareGroupStateResult> deleteState(DeleteShareGro
    * @param request ReadShareGroupStateSummaryParameters
    * @return ReadShareGroupStateSummaryResult
    */
-  public CompletableFuture<ReadShareGroupStateSummaryResult> readSummary(ReadShareGroupStateSummaryParameters request) {
+  public CompletableFuture<ReadShareGroupStateSummaryResult> readSummary(ReadShareGroupStateSummaryParameters request) throws IllegalArgumentException {
     throw new RuntimeException("not implemented");
   }
+
+  private static void validate(WriteShareGroupStateParameters params) {
+    String prefix = "Write share group parameters";
+    if (params == null) {
+      throw new IllegalArgumentException(prefix + " cannot be null.");
+    }
+    if (params.groupTopicPartitionData() == null) {
+      throw new IllegalArgumentException(prefix + " data cannot be null.");
+    }
+
+    validateGroupTopicPartitionData(prefix, params.groupTopicPartitionData());
+  }
+
+  private static void validate(ReadShareGroupStateParameters params) {
+    String prefix = "Read share group parameters";
+    if (params == null) {
+      throw new IllegalArgumentException(prefix + " cannot be null.");
+    }
+    if (params.groupTopicPartitionData() == null) {
+      throw new IllegalArgumentException(prefix + " data cannot be null.");
+    }
+
+    validateGroupTopicPartitionData(prefix, params.groupTopicPartitionData());
+  }
+
+  private static void validateGroupTopicPartitionData(String prefix, GroupTopicPartitionData<? extends PartitionIdData> data) {
+    String groupId = data.groupId();
+    if (groupId == null || groupId.isEmpty()) {
+      throw new IllegalArgumentException(prefix + " groupId cannot be null or empty.");
+    }
+
+    List<? extends TopicData<? extends PartitionIdData>> topicsData = data.topicsData();
+    if (isEmpty(topicsData)) {
+      throw new IllegalArgumentException(prefix + " topics data cannot be null or empty.");
+    }
+
+    for (TopicData<? extends PartitionIdData> topicData : topicsData) {
+      if (topicData.topicId() == null) {
+        throw new IllegalArgumentException(prefix + " topicId cannot be null.");
+      }
+      if (isEmpty(topicData.partitions())) {
+        throw new IllegalArgumentException(prefix + " partitions cannot be null or empty.");
+      }
+      for (PartitionIdData partitionData : topicData.partitions()) {
+        if (partitionData.partition() == -1) {
```

Comment:
Should we also validate for other non-negative values for topic partition, i.e. use partitionData.partition() < 0 instead of partitionData.partition() == -1 inside the if check ?